src/ui/pages/AssetInfo/AssetResources.tsx

```diff
@@ -0,0 +1,365 @@
+      ) || implementations[0]
+    );
+  }, [implementations, assetFullInfo.extra.mainChain]);
+
+  const resourcesById = useMemo(() => {
+    return assetFullInfo.extra.relevantResources.reduce((acc, resource) => {
+      acc[resource.name] = resource;
+      return acc;
+    }, {} as Record<string, AssetResource>);
+  }, [assetFullInfo.extra.relevantResources]);
+
+  return (
+    <>
+      <HStack gap={8} justifyContent="space-between">
+        <HStack gap={8}>
+          {resourcesById[TWITTER_ID] ? (
+            <ResourseButton
+              resource={resourcesById[TWITTER_ID]}
+              icon={<XIcon style={{ width: 20, height: 20 }} />}
+            />
+          ) : null}
+          {resourcesById[WARPCAST_ID] ? (
+            <ResourseButton
+              resource={resourcesById[WARPCAST_ID]}
+              icon={<WarpcastIcon style={{ width: 20, height: 20 }} />}
+            />
+          ) : null}
+          <ResourseButton
+            resource={
+              resourcesById[DEXSCREENER_ID] || {
+                name: DEXSCREENER_ID,
+                displayableName: 'Dexscreener',
+                iconUrl: '',
+                url: `https://dexscreener.com/search?q=${assetFullInfo.fungible.symbol}`,
+              }
+            }
+            icon={<DexscreenerIcon style={{ width: 20, height: 20 }} />}
+          />
+        </HStack>
+        <HStack gap={4}>
+          {mainImplementation ? (
+            <MainImplementationButton implementation={mainImplementation} />
+          ) : null}
+          {implementations.length ? (
+            <Button
+              kind="regular"
+              size={36}
+              onClick={() => dialogRef.current?.showModal()}
+              style={{
+                padding: 8,
+                ['--button-background' as string]: 'var(--neutral-200)',
+                ['--button-background-hover' as string]: 'var(--neutral-300)',
+              }}
+            >
+              <DownIcon style={{ width: 20, height: 20 }} />
+            </Button>
+          ) : null}
+        </HStack>
+      </HStack>
+      <CenteredDialog
+        ref={dialogRef}
+        containerStyle={{ backgroundColor: 'var(--white)', padding: 0 }}
+        renderWhenOpen={() => (
+          <>
+            <HStack
+              gap={0}
+              alignItems="center"
+              style={{
+                position: 'sticky',
+                top: 0,
+                padding: '16px 8px 0',
+                gridTemplateColumns: '36px 1fr 36px',
+                backdropFilter: 'blur(8px)',
+              }}
+            >
+              <Button
+                kind="ghost"
+                value="cancel"
+                size={36}
+                style={{ width: 36, padding: 8 }}
+                onClick={() => dialogRef.current?.close()}
+              >
+                <ArrowLeftIcon style={{ width: 20, height: 20 }} />
+              </Button>
```

Comment:
Technically html dialogs are designed like this:
```suggestion
              <form method="dialog" onSubmit={(event) => event.stopPropagation()}>
                <Button
                  kind="ghost"
                  value="cancel"
                  size={36}
                  style={{ width: 36, padding: 8 }}
                >
                  <ArrowLeftIcon style={{ width: 20, height: 20 }} />
                </Button>
              </form>
```

Not crucial, but this makes this component decoupled from the `dialogRef` variable

Reply:
Interesting!